presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1115,14 +1117,14 @@ public ConnectorTableLayout getTableLayout(ConnectorSession session, ConnectorTa
                 ImmutableList.of());
     }
 
-    private static TupleDomain<ColumnHandle> createPredicate(List<ColumnHandle> partitionColumns, List<HivePartition> partitions)
+    @VisibleForTesting
+    static TupleDomain<ColumnHandle> createPredicate(List<ColumnHandle> partitionColumns, List<HivePartition> partitions)
     {
         if (partitions.isEmpty()) {
             return TupleDomain.none();
         }
 
-        Map<ColumnHandle, Domain> domains = new HashMap<>(partitionColumns.size());
-
+        Multimap<ColumnHandle, NullableValue> partitionToValues = HashMultimap.create();
```

Comment:
Is this cleaner than making multiple passes through the list of partitions, though? It's probably faster.